src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2025, Hewlett Packard Enterprise Development LP */
+
+/*
+ * util_sds_check.c -- unit test for shutdown_state_check function
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "unittest.h"
+#include "ut_pmem2.h"
+#include "shutdown_state.h"
+#include "set.h"
+
+int os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush);
+/*
+ * os_part_deep_common -- XXX temporally workaround until we will have pmem2
+ *		integrated with common
+ */
+int
+os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush)
+{
+	return 0;
+}
+
+/* Test with curr_sds having non-zero values and pool_sds being zeroed out. */
```

Comment:
```suggestion
/* The persistent SDS turned out to be zeroed out whereas based on the pool structure it should be not. */
```

Can we stop for a minute and discuss how it could have happened? It sounds a little suspicious.

@grom72 ?

Reply:
Done.

Reply:
As we agreed offline. I am considering adding:

- a comment to the source code and
- a warning message just to keep track of this event which should happen only once and in very peculiar conditions.

Please see both below.

```diff
diff --git a/src/common/shutdown_state.c b/src/common/shutdown_state.c
index d6b8979f7..29755362d 100644
--- a/src/common/shutdown_state.c
+++ b/src/common/shutdown_state.c
@@ -166,8 +166,15 @@ shutdown_state_check(struct shutdown_state *curr_sds,
 {
 	LOG(3, "curr_sds %p, pool_sds %p", curr_sds, pool_sds);
 
+	/**
+	 * This is likely to occur only when the pool is being opened for
+	 * the first time after the SHUTDOWN_STATE feature has been enabled on
+	 * the pool, for example, via (lib)pmempool.
+	 * Please do not confuse this with establishing SDS during creation.
+	 */
 	if (util_is_zeroed(pool_sds, sizeof(*pool_sds)) &&
 			!util_is_zeroed(curr_sds, sizeof(*curr_sds))) {
+		CORE_LOG_WARNING("zeroed out - SDS will be reinitialized");
 		shutdown_state_reinit(curr_sds, pool_sds, rep);
 		return 0;
 	}
```